bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/PiHoleHandler.java

```diff
@@ -0,0 +1,292 @@
+            try {
+                hostname = new URI(config.hostname);
+            } catch (URISyntaxException e) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                        "@token/handler.init.invalidHostname[\"" + config.hostname + "\"]");
+                return;
+            }
+            if (config.token.isEmpty()) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR, "@token/handler.init.noToken");
+                return;
+            }
+            adminService = new JettyAdminService(config.token, hostname, httpClient);
+        }
+
+        if (config.refreshIntervalSeconds <= 0) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "@text/handler.init.wrongInterval");
+            return;
+        }
+        scheduledFuture = scheduler.scheduleWithFixedDelay(this::update, 0, config.refreshIntervalSeconds, SECONDS);
+
+        // do not set status here, the background task will do it.
+    }
+
+    private void update() {
+        var local = adminService;
+        if (local == null) {
+            return;
+        }
+
+        // this block can be called from at least 2 threads
+        // check disableBlocking method
+        synchronized (lock) {
+            try {
+                logger.debug("Refreshing DnsStatistics from PiHole");
+                local.summary().ifPresent(statistics -> dnsStatistics = statistics);
+                refresh();
+                updateStatus(ONLINE);
+            } catch (Exception e) {
+                logger.debug("Error occurred when refreshing DnsStatistics from PiHole", e);
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getLocalizedMessage());
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refresh();
+            return;
+        }
+
+        if (DISABLE_ENABLE_CHANNEL.equals(channelUID.getId())) {
+            if (command instanceof StringType stringType) {
+                var value = DisableEnable.valueOf(stringType.toString());
+                try {
+                    switch (value) {
+                        case DISABLE -> disableBlocking(0);
+                        case FOR_10_SEC -> disableBlocking(10);
+                        case FOR_30_SEC -> disableBlocking(30);
+                        case FOR_5_MIN -> disableBlocking(MINUTES.toSeconds(5));
+                        case ENABLE -> enableBlocking();
+                    }
+                } catch (ExecutionException | InterruptedException | TimeoutException ex) {
+                    logger.debug("Cannot invoke {} on channel {}", value, channelUID, ex);
+                    updateStatus(OFFLINE, COMMUNICATION_ERROR, ex.getLocalizedMessage());
+                }
+            }
+        }
+    }
+
+    private void refresh() {
+        var localDnsStatistics = dnsStatistics;
+        if (localDnsStatistics == null) {
+            return;
```

Comment:
If this is not the normal behaviour, leave a message by log.

Reply:
It can happen at the startup of binding at it is not not normal behavior. 

From what I see logging should be kept to minimal